src/modules/simulation/simulator_ignition_bridge/SimulatorIgnitionBridge.cpp

```diff
@@ -76,14 +76,14 @@ int SimulatorIgnitionBridge::init()
 	if (!_model_pose.empty()) {
 		PX4_INFO("Requested Model Position: %s", _model_pose.c_str());
 
-		std::vector<float> model_pose_v;
+		std::vector<double> model_pose_v;
 
 		std::stringstream ss(_model_pose);
 
 		while (ss.good()) {
 			std::string substr;
 			std::getline(ss, substr, ',');
-			model_pose_v.push_back(std::stof(substr));
```

Comment:
Whas this intentional? I don't think this should be part of the PR

Reply:
It was intentional- the code was not compiling until I made that change. Perhaps it shouldn't be part of this specific PR (the focus is on the airframe), but I think it should be changed if it is a common problem.

Reply:
This is the error I saw: 
`PX4-Autopilot/src/modules/simulation/simulator_ignition_bridge/SimulatorIgnitionBridge.cpp:95:19: fatal error: implicit conversion increases floating-point precision: '__gnu_cxx::__alloc_traits<std::allocator<float>, float>::value_type' (aka 'float') to 'double' [-Wdouble-promotion]
                position->set_x(model_pose_v[0]);
`

Reply:
> but I think it should be changed if it is a common problem.

I don't think it is a common problem. Could you explain a bit more which environment you are using?

Reply:
As far as I can tell, it's the latest commit of the PX4 firmware, with the Gazebo modified to include the new lift-drag plugin.

Reply:
@karthik-s-feather Just to clarify, you are saying that you can't build firmware even without the advanced lift drag changes and need to modify the `SimulatorIgnitionBridge`?

Which operating system are you using? The latest commit on firmware should build with gazebo according to the CI: https://github.com/PX4/PX4-Autopilot/actions/runs/3045503156/jobs/4907151073

Reply:
@Jaeyoung-Lim It seems so. I just attempted to build the main branch with gazebo, without my changes, and it still failed with the above error. 
I'm on Ubuntu 20.04.5 LTS.

Reply:
Here's how I cloned the repo: 
`git clone git@github.com:PX4/PX4-Autopilot.git
cd PX4-Autopilot/
git submodule update --init --recursive`
After that, I tried the `make` commands listed above and they failed.

Reply:
@karthik-s-feather It is anyway an error coming from a warning (`-Wdouble-promotion`), so I would still keep it out of the PR. If you think this is something worth fixing, please address it in a different PR. I just tested on my system and it works fine for me

Reply:
Got it. Does the double promotion warning show up for you? If so, it could be something in my build settings.